partner_create_by_vat/models/res_partner.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# ©  2015 Forest and Biomass Services Romania
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+from openerp.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _get_vies_data(self, vat):
+        try:
+            from stdnum.eu.vat import check_vies
+        except:
+            ValidationError(_("There was an error importing check_vies "
+                              "method from python stdnum."))
+        res = {}
+        vat = vat.strip().upper()
+        vat_country, vat_number = self._split_vat(vat)
+        result = check_vies(vat)
+        # Check if partner is listed on Vies
+        if result.name is not None:
```

Comment:
(non-blocking) What about change the style of program to avoid `else` section and reduce the [cyclomatic complexit](https://en.wikipedia.org/wiki/Cyclomatic_complexity)

Maybe 
```python
# Check if partner is listed on Vies
if result.name is None:
    raise ValidationError(_("The partner is not listed on Vies "
                                     "Webservice."))
if result.name == '---':
    res['vat_subjected'] = result.valid
    raise ValidationError(_("The partner doesn't have the name "
                                         "and address listed on Vies "
                                         "Webservice."))

# Get country by country code
country = self.env['res.country'].search(
    [('code', 'ilike', vat_country)])
new_name = result.name.upper()
if result.address != '---':
    new_address = result.address.replace(
        '\n', ' ').replace('\r', '').title()
res.update({
    'name': new_name,
    'vat': vat,
    'street': new_address,
    'country_id': country and country[0].id,
    'vat_subjected':  result.valid
return res
```


NOTE: This code is a example don't use as a real code.